tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -20,16 +21,33 @@ internal class LambdaCommon
         private const string DefaultJson = "{}";
         private const double ServerlessMaxWaitingFlushTime = 3;
 
-        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder)
+        internal static CallTargetState StartInvocation<TArg>(TArg payload, ILambdaExtensionRequest requestBuilder, IDictionary<string, string> context)
         {
             var json = SerializeObject(payload);
-            NotifyExtensionStart(requestBuilder, json);
+            NotifyExtensionStart(requestBuilder, json, context ?? new Dictionary<string, string>());
             return new CallTargetState(CreatePlaceholderScope(Tracer.Instance, requestBuilder));
         }
 
         internal static CallTargetState StartInvocationWithoutEvent(ILambdaExtensionRequest requestBuilder)
         {
-            return StartInvocation(DefaultJson, requestBuilder);
+            return StartInvocation(DefaultJson, requestBuilder, null);
+        }
+
+        internal static CallTargetState StartInvocationOneParameter<TArg>(TArg eventOrContext, ILambdaExtensionRequest requestBuilder)
+        {
+            if (eventOrContext.GetType().ToString().Contains("LambdaContext"))
+            {
+                var dict = GetPropertyValue(GetPropertyValue(eventOrContext, "ClientContext"), "Custom");
```

Comment:
If this is in the hot path, we should consider using our duck typing feature to avoid using reflection, which is infamously slow.

Reply:
implemented!